python/foglamp/plugins/south/modbustcp/modbustcp.py

```diff
@@ -158,54 +156,76 @@ def plugin_poll(handle):
     try:
         global mbus_client
         if mbus_client is None:
-            source_ip = handle['sourceIpAddress']['value']
             try:
+                source_ip = handle['sourceIpAddress']['value']
                 source_port = int(handle['sourcePort']['value'])
             except:
                 raise ValueError
+
             mbus_client = ModbusTcpClient(host=source_ip, port=source_port)
             _LOGGER.info('Modbus TCP started on IP %s', source_ip)
 
-        """ TODO: use modbus_map = handle['entitiesMap']['value'] dict to read
-        
+        """ 
         read_coils(self, address, count=1, **kwargs)  
         read_discrete_inputs(self, address, count=1, **kwargs)
         read_holding_registers(self, address, count=1, **kwargs)
         read_input_registers(self, address, count=1, **kwargs)
-        
+
             - address: The starting address to read from
             - count: The number of coils / discrete or registers to read
             - unit: The slave unit this request is targeting
-            
+
             On TCP/IP, the MODBUS server is addressed using its IP address; therefore, the MODBUS Unit Identifier is useless. 
             The value 0xFF has to be used. When addressing a MODBUS server connected directly to a TCP/IP network, it’s 
             recommended not using a significant MODBUS slave address in the "Unit Identifier" field.  
             In the event of a re-allocation of the IP addresses within an automated system and if a IP address previously
             assigned to a MODBUS server is then assigned to a gateway, using a significant slave address may cause trouble
             because of a bad routing by the gateway. Using a non-significant slave address, the gateway will simply discard
             the MODBUS PDU with no trouble. 0xFF is recommended for the "Unit Identifier" as non-significant value. 
-            
+
             Remark : The value 0 is also accepted to communicate directly to a MODBUS TCP device.
-. 
-       
-        """
 
-        # Specify which register number to monitor and how many registers to read
-        register_number = 0
+        """
+        unit_id = 0
+        modbus_map = json.loads(handle['entitiesMap']['value'])
+        readings = {}
+
+        # Read coils
+        coils_address_info = modbus_map['coils']
+        number_of_coils_to_read = 1
+        if len(coils_address_info) > 0:
+            for k, address in coils_address_info.items():
+                coil = mbus_client.read_coils(address, number_of_coils_to_read, unit_id)

```

Comment:
this will never work as unit_id is **kwargs, so please use `unit=UNIT_ID` let's set UNIT_ID at module level as `UNIT = 0x1` or `UNIT = 0x0`